pkg/operator/resource/resourceapply/rbac.go

```diff
@@ -37,13 +32,21 @@ func ApplyClusterRole(ctx context.Context, client rbacclientv1.ClusterRolesGette
 	existingCopy := existing.DeepCopy()
 
 	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
-	contentSame := equality.Semantic.DeepEqual(existingCopy.Rules, required.Rules)
-	if contentSame && !*modified {
+	rulesContentSame := equality.Semantic.DeepEqual(existingCopy.Rules, required.Rules)
+	aggregationRuleContentSame := equality.Semantic.DeepEqual(existingCopy.AggregationRule, required.AggregationRule)
+
+	if aggregationRuleContentSame && rulesContentSame && !*modified {
 		return existingCopy, false, nil
 	}
 
-	existingCopy.Rules = required.Rules
-	existingCopy.AggregationRule = nil
+	// The control plane controller that reconciles ClusterRoles
+	// overwrites any values that are manually specified in the rules field of an aggregate ClusterRole.
+	// As such skip reconciling on the Rules field when the AggregationRule filed changes.
+	if !aggregationRuleContentSame {
+		existingCopy.AggregationRule = required.AggregationRule
+	} else {
+		existingCopy.Rules = required.Rules
+	}
```

Comment:
Should this actually be based on whether the aggregation rule is set or not? You never want to set the `Rules` for an aggregation rule right?

This looks to me like it will flip flop updating the Rules because they differ, when the aggregation rule doesn't change?

Reply:
I re-read the code in CVO and yes, I agree with you.
I've updated to now check on `required.AggregationRule == nil`.